src/table/internal.tsx

```diff
@@ -446,19 +446,26 @@ const InternalTable = React.forwardRef(
                       </tr>
                     ) : (
                       allItems.map((item, rowIndex) => {
-                        const firstVisible = rowIndex === 0;
-                        const lastVisible = rowIndex === allItems.length - 1;
-                        const isEven = rowIndex % 2 === 0;
-                        const isSelected = !!selectionType && isItemSelected(item);
-                        const isPrevSelected =
-                          !!selectionType && !firstVisible && isItemSelected(allItems[rowIndex - 1]);
-                        const isNextSelected =
-                          !!selectionType && !lastVisible && isItemSelected(allItems[rowIndex + 1]);
+                        const isFirstRow = rowIndex === 0;
+                        const isLastRow = rowIndex === allItems.length - 1;
+                        const sharedCellProps = {
+                          isFirstRow,
+                          isLastRow,
+                          isVisualRefresh,
+                          isSelected: !!selectionType && isItemSelected(item),
```

Comment:
Nitpick: `!!selectionType` could be refactored as `hasSelection` in these 3 lines, similarly to how it was refactored in line 477 / 484.